src/app/src/components/AnimatedArcsAndMap/AnimatedArcsOverStates.tsx

```diff
@@ -0,0 +1,161 @@
+import { useEffect, useRef, useCallback } from 'react';
+import { useMap } from 'react-leaflet';
+
+import L from 'leaflet';
+import '@elfalem/leaflet-curve';
+import { StateGeometry, StateProperties } from '../states.geojson';
+
+import StatesLayer from '../StatesLayer';
+import {
+    MonthlySpendingOverTime,
+    MonthlySpendingOverTimeByState,
+} from '../../types/api';
+import { MONTHLY_TIME_DURATION, DC_CENTER } from '../../constants';
+
+function getBezierOffsetLatLng(end: L.LatLngTuple): L.LatLngTuple {
+    // Modified Bezier curve function from https://gist.github.com/ryancatalani/6091e50bf756088bf9bf5de2017b32e6
+    const start: L.LatLngTuple = DC_CENTER;
+    var offsetX = end[1] - start[1],
+        offsetY = end[0] - start[0];
+
+    var r = Math.sqrt(Math.pow(offsetX, 2) + Math.pow(offsetY, 2)),
+        theta = Math.atan2(offsetY, offsetX);
+
+    var thetaOffset = 3.14 / 10;
+
+    var r2 = r / 2 / Math.cos(thetaOffset),
+        theta2 = theta + thetaOffset;
+
+    var midpointX = r2 * Math.cos(theta2) + start[1],
+        midpointY = r2 * Math.sin(theta2) + start[0];
+
+    return [midpointY, midpointX];
+}
+
+export default function AnimatedArcsOverStates({
+    animationEnabled,
+    spending,
+    totalTimeSteps,
+}: {
+    animationEnabled: boolean;
+    spending: MonthlySpendingOverTimeByState;
+    totalTimeSteps: number;
+}) {
+    const map = useMap();
+    const arcPathsReference = useRef<{ shape_code: string; curve: L.Curve }[]>(
```

Comment:
Maybe this would make more sense as a `Record<string, L.Curve>`? That way you can check if a state curve exists by doing `shape_code in arcPathsReference.current` and access the curve with `arcPathsReference.current[shape_code]`.